src/transformers/generation_utils.py

```diff
@@ -39,6 +40,287 @@
 logger = logging.get_logger(__name__)
 
 
+@dataclass
+class GreedySearchDecoderOnlyOutput(ModelOutput):
+    """
+    Base class for outputs of encoder-decoder generation models using greedy search.
```

Comment:
This is the same thing as `GreedySearchEncoderDecoderOutput` which seems wrong. This line should mention the "decoder only" part of the name (and maybe quickly explain how it's different from the encoder-decoder)